src/GafferSceneUIModule/SetEditorBinding.cpp

```diff
@@ -0,0 +1,275 @@
+#include "Gaffer/PathFilter.h"
+
+#include "IECorePython/RefCountedBinding.h"
+
+#include "IECore/StringAlgo.h"
+
+#include "boost/algorithm/string/predicate.hpp"
+#include "boost/python/suite/indexing/container_utils.hpp"
+
+using namespace std;
+using namespace boost::python;
+using namespace IECore;
+using namespace IECorePython;
+using namespace Gaffer;
+using namespace GafferUI;
+using namespace GafferScene;
+using namespace GafferSceneUI;
+
+namespace
+{
+
+const InternedString g_setNamePropertyName( "setPath:setName" );
+const InternedString g_memberCountPropertyName( "setPath:memberCount" );
+
+ConstStringDataPtr g_emptySetIcon = new StringData( "emptySet.png" );
+ConstStringDataPtr g_populatedSetIcon = new StringData( "populatedSet.png" );
+ConstStringDataPtr g_setFolderIcon = new StringData( "setFolder.png" );
+
+class SetNameColumn : public StandardPathColumn
+{
+
+	public :
+
+		IE_CORE_DECLAREMEMBERPTR( SetNameColumn )
+
+		SetNameColumn()
+			:	StandardPathColumn( "Name", "name", GafferUI::PathColumn::SizeMode::Stretch )
+		{
+		}
+
+		CellData cellData( const Gaffer::Path &path, const IECore::Canceller *canceller ) const override
+		{
+			CellData result = StandardPathColumn::cellData( path, canceller );
+
+			const auto setName = runTimeCast<const IECore::StringData>( path.property( g_setNamePropertyName, canceller ) );
+			if( !setName )
+			{
+				result.icon = g_setFolderIcon;
+			}
+			else
+			{
+				const auto memberCount = runTimeCast<const IECore::IntData>( path.property( g_memberCountPropertyName, canceller ) );
+				if( memberCount )
+				{
+					result.icon = memberCount->readable() > 0 ? g_populatedSetIcon : g_emptySetIcon;
+				}
+			}
+
+			return result;
+		}
+
+};
+
+//////////////////////////////////////////////////////////////////////////
+// SetEditorSearchFilter - filters based on a match pattern. This
+// removes non-leaf paths if all their children have also been
+// removed by the filter.
+//////////////////////////////////////////////////////////////////////////
+
+class SetEditorSearchFilter : public Gaffer::PathFilter
+{
+
+	public :
+
+		IE_CORE_DECLAREMEMBERPTR( SetEditorSearchFilter )
+
+		SetEditorSearchFilter( IECore::CompoundDataPtr userData = nullptr )
+			:	PathFilter( userData )
+		{
+		}
+
+		void setMatchPattern( const string &matchPattern )
+		{
+			if( m_matchPattern == matchPattern )
+			{
+				return;
+			}
+			m_matchPattern = matchPattern;
+			m_wildcardPattern = IECore::StringAlgo::hasWildcards( matchPattern ) ? matchPattern : "*" + matchPattern + "*";
+
+			changedSignal()( this );
+		}
+
+		const string &getMatchPattern() const
+		{
+			return m_matchPattern;
+		}
+
+		void doFilter( std::vector<PathPtr> &paths, const IECore::Canceller *canceller ) const override
+		{
+			if( m_matchPattern.empty() || paths.empty() )
+			{
+				return;
+			}
+
+			paths.erase(
+				std::remove_if(
+					paths.begin(),
+					paths.end(),
+					boost::bind( &SetEditorSearchFilter::remove, this, ::_1 )
```

Comment:
We should be able to use lambdas instead of `boost::bind()` now.

Reply:
Swapped `boost::bind()` for lambdas in 65f28fe, much better!